src/analysis.py

```diff
@@ -274,7 +275,7 @@ def optimize_model(train_df, param_grid=None, output_dir="", fileprefix=""):
     numeric_features = ['appearances', 'year', 'name_len', 'appear_per_yr']
     categorical_features = ['id', 'eye', 'hair', 'publisher', 'sex']
     drop_features = ['name', 'gsm', 'first_appearance', 'first_name', 'last_name']
-    binary_features = ['is_common', 'last_name']
+    binary_features = ['is_common', 'last_name', 'has_last_name']
```

Comment:
binary_features = ['is_common', 'has_last_name'], my final csv do not have 'last_name' column.

Reply:
Good catch!

Reply:
Removed - think this might have fixed binary specific OHE.
Thanks!